### PR TITLE
amp-iframe guideline to prefer existing components

### PR DESCRIPTION
**Pull request by ericlindley-g**
Thought it would be good to document this somewhere, but want to make sure it's accurate and useful. Not sure who is the best person to review, but /to @cramforce  as a first guess, given the component history

### CODE REVIEW

File: extensions/amp-iframe/amp-iframe.md [hunk 1]
```diff
@@ -208,3 +208,13 @@ Iframes are identified as tracking/analytics iframes if they appear to serve no
 ## Validation
 
 See [amp-iframe rules](https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/0.1/validator-amp-iframe.protoascii) in the AMP validator specification.
+
+## Guideline: prefer existing AMP components to `amp-iframe`
+
+`amp-iframe` should be considered a fallback if the required user experience is not possible by other means in AMP. This is because there are many benefits to using a component tailored for a specific use-case instead, such as
+
+- Better resource management and performance
+- Increased loading priority (`amp-iframe` loads last, because AMP doesn't know what's inside of it)
```
Comment (cramforce) [inline extensions/amp-iframe/amp-iframe.md]: Not correct

File: extensions/amp-iframe/amp-iframe.md [hunk 2]
```diff
@@ -208,3 +208,13 @@ Iframes are identified as tracking/analytics iframes if they appear to serve no
...
+- Custom components can provide built-in placeholder images in some cases. This means getting, say, the right video thumbnail before a video loads, and reduces the coding effort to add a placeholder manually.
+- Built-in resize requests. This means that iframe content with unpredictable size can more often appear to the user as if it were native to the page, rather than in a scrollable frame
```
Comment (cramforce) [inline extensions/amp-iframe/amp-iframe.md]: Built-in resizing.

File: extensions/amp-iframe/amp-iframe.md [hunk 3]
```diff
@@ -208,3 +208,13 @@ Iframes are identified as tracking/analytics iframes if they appear to serve no
 ## Validation
 
 See [amp-iframe rules](https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/0.1/validator-amp-iframe.protoascii) in the AMP validator specification.
+
+## Guideline: prefer existing AMP components to `amp-iframe`
```
Comment (cramforce) [inline extensions/amp-iframe/amp-iframe.md]: s/existing/specific

Please add link to list of components.

File: extensions/amp-iframe/amp-iframe.md (hunk 1 above)
Comment (ericlindley-g) [inline extensions/amp-iframe/amp-iframe.md]: Got it—is there anything we can say around better loading priority for specific components v. amp-iframe, or should this line just be deleted?

Comment (cramforce) [inline extensions/amp-iframe/amp-iframe.md]: Just delete it.

File: extensions/amp-iframe/amp-iframe.md (hunk 3 above)
Comment (ericlindley-g) [inline extensions/amp-iframe/amp-iframe.md]: Done

File: extensions/amp-iframe/amp-iframe.md (hunk 1 above)
Comment (ericlindley-g) [inline extensions/amp-iframe/amp-iframe.md]: Done

File: extensions/amp-iframe/amp-iframe.md (hunk 2 above)
Comment (ericlindley-g) [inline extensions/amp-iframe/amp-iframe.md]: Done

**Comment by ericlindley-g** [comment]
Thanks for the feedback—should be updated to address the comments now

File: extensions/amp-iframe/amp-iframe.md [hunk 4]
```diff
@@ -208,3 +208,12 @@ Iframes are identified as tracking/analytics iframes if they appear to serve no
 ## Validation
 
 See [amp-iframe rules](https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/0.1/validator-amp-iframe.protoascii) in the AMP validator specification.
+
+## Guideline: prefer [specific AMP components](https://github.com/ampproject/amphtml/tree/master/extensions) to `amp-iframe`
```
Comment (cramforce) [inline extensions/amp-iframe/amp-iframe.md]: When this goes into the docs site, please check that the link there points to the right destination. CC @pbakaus

**Pull request merged**